commons/src/tracing.rs

```diff
@@ -0,0 +1,85 @@
+//! Tracing service.
+
+use opentelemetry::api::{
+    Carrier, HttpTextFormat, Key, Provider, Span, SpanContext, TraceContextPropagator,
+};
+use opentelemetry::{global, sdk};
+use opentelemetry_jaeger::{Exporter, Process};
+
+use actix_web::dev::ServiceRequest;
+use actix_web::http;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+
+/// init_tracer sets up Jaeger tracer
+pub fn init_tracer(name: &'static str) -> thrift::Result<()> {
+    let exporter = Exporter::builder()
+        .with_agent_endpoint("127.0.0.1:6831".parse().unwrap())
```

Comment:
I do think we should expose this as a configuration option, e.g. as a `Option<String>` to make it really simple. if that is `None`, we just return `Ok(())` here immediately without setting up anything. Would that result in any error down the line?

Reply:
Added PE/GB settings to customize the endpoints. 

Not entirely sure if "instrument" calls won't crash if provider was not initialized - I'll give it a try

Reply:
Update: both GB and PE work as expected if `init_tracer` is omitted

Should we hide it behind the feature flag / config option? I'd prefer not to, so that Jeager could seamlessly receive traces without Cincy restart or reconfiguration

Reply:
Is it likely that we need to plug-in Jaeger in that manner? I would assume we'd either know it's running and configure its address appropriately, or we know it's now running and skip the config option.

Reply:
`now` -> `not` in "we know it's now running"?

Okay, lets go with that. If the setting is set its configured, if not - the function exits

Reply:
Implemented in 8fe8fee